### PR TITLE
chore: add e2e test running on AWS infra

### PR DESCRIPTION
**Pull request by Unix4ever**
Using `capi-utils` module to set up cluster in AWS using current version
of `control-plane-provider`.
Only validate health in the tests.

Signed-off-by: Artem Chernyshev <artem.chernyshev@talos-systems.com>

### CODE REVIEW

File: hack/test/e2e-aws.sh [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+    ${REGISTRY_MIRROR_FLAGS} \
+    --crashdump
+  ${TALOSCTL} config nodes 10.5.0.2
+  ${TALOSCTL} kubeconfig -f ${TMP}/kubeconfig
+
+  export KUBECONFIG=${TMP}/kubeconfig
+}
+
+function create_ami {
+  image_name=talos-e2e-cacppt-${TALOS_VERSION}
```
Comment (smira) [inline hack/test/e2e-aws.sh:108]: we shouldn't need to upload AMIs, as we're running some released version of Talos

https://www.talos.dev/docs/v0.12/cloud-platforms/aws/#official-ami-images

File: internal/integration/integration_test.go [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+	env := func(key, def string) string {
+		val := os.Getenv(key)
+		if val != "" {
+			return val
+		}
+
+		return def
+	}
+
+	providerType := env("PROVIDER", "aws:v0.6.7")
```
Comment (smira) [inline internal/integration/integration_test.go:59]: should this be part of the `SetupSuite()` ? as we should do it once for all the tests I guess

File: go.mod [hunk 3]
```diff
@@ -2,19 +2,32 @@ module github.com/talos-systems/cluster-api-control-plane-provider-talos
...
 go 1.16
 
+replace (
+	// keep older versions of k8s.io packages to keep compatiblity with cluster-api
+	k8s.io/api v0.21.3 => k8s.io/api v0.20.5
+	k8s.io/api-server v0.21.3 => k8s.io/api-server v0.20.5
+	k8s.io/apimachinery v0.21.3 => k8s.io/apimachinery v0.20.5
+	k8s.io/client-go v0.21.3 => k8s.io/client-go v0.20.5
+
+	sigs.k8s.io/cluster-api v0.3.20 => sigs.k8s.io/cluster-api v0.3.9
```
Comment (smira) [inline go.mod:12]: :shrug: I'm afraid of those bumps and crazy overrides... I know it comes from `capi-utils` probably

**Review by smira — COMMENTED** [review]
:cool: :trophy:

File: go.mod (hunk 3 above)
Comment (Unix4ever) [reply go.mod:12]: I hope we'll get rid of it as soon as we migrate to v1alpha4

**Review by smira — APPROVED** [review]
:1st_place_medal:

**Pull request merged**